packages/odf/components/storage-consumers/onboarding-modal.tsx

```diff
@@ -91,41 +152,204 @@ export const ClientOnBoardingModal: ClientOnBoardingModalProps = ({
                 </Trans>
               </Alert>
             )}
-          </FlexItem>
-          {!error && !loading && (
-            <FlexItem>
-              <Button
-                type="button"
-                onClick={onCopyToClipboard}
-                variant="link"
-                className="pf-m-link--align-left odf-onboarding-modal__clipboard"
-              >
-                <ClipboardIcon />
-                {t('Copy to clipboard')}
-              </Button>
-            </FlexItem>
-          )}
-          <FlexItem>
-            <Text component={TextVariants.h6}>
-              {t('How to use this token?')}
-            </Text>
-            <Text>
-              {t(
-                'An onboarding token is needed to connect an additional OpenShift cluster to a Data Foundation deployment. Copy the generated token and use it for deploying Data Foundation client operator on your OpenShift cluster.'
-              )}
-            </Text>
-          </FlexItem>
-          <FlexItem>
-            <StatusIconAndText
-              title={t(
-                'This token is valid for 48 hours and can only be used once.'
-              )}
-              icon={<BlueInfoCircleIcon />}
-              className="text-muted"
-            />
-          </FlexItem>
-        </Flex>
-      </ModalBody>
+            <StorageQuotaBody quota={quota} setQuota={setQuota} />
+          </ModalBody>
+          <ModalFooter inProgress={inProgress}>
+            <Flex direction={{ default: 'row' }}>
+              <FlexItem>
+                <Button
+                  key="cancel"
+                  variant="secondary"
+                  onClick={closeModal}
+                  data-test-id="modal-cancel-action"
+                >
+                  {t('Cancel')}
+                </Button>
+              </FlexItem>
+              <FlexItem>
+                <Button
+                  key="save"
+                  data-test="modal-submit-action"
+                  data-test-id="confirm-action"
+                  variant="primary"
+                  onClick={() => generateToken()}
+                  isDisabled={inProgress}
+                  isLoading={inProgress}
+                >
+                  {inProgress ? t('Generating token') : t('Generate token')}
+                </Button>
+              </FlexItem>
+            </Flex>
+          </ModalFooter>
+        </>
+      )}
     </Modal>
   );
 };
+
+type StorageQuotaBodyProps = {
+  quota: StorageQuota;
+  setQuota: React.Dispatch<React.SetStateAction<StorageQuota>>;
+  initialQuota?: StorageQuota;
+  capacityInfo?: React.ReactNode;
+};
+
+export const StorageQuotaBody: React.FC<StorageQuotaBodyProps> = ({
+  quota,
+  setQuota,
+  initialQuota,
+  capacityInfo,
+}) => {
+  const { t } = useCustomTranslation();
+  const [showCustomQuotaInput, setShowCustomQuotaInput] = React.useState(
+    !isUnlimitedQuota(quota)
+  );
```

Comment:
we can determine it during render itself... no need for a new state...
```suggestion
    const shouldShowCustomQuotaInput = !isUnlimitedQuota(quota);
```

Reply:
Done.